httpclient/src/main/java/org/apache/http/impl/client/DefaultRedirectStrategy.java

```diff
@@ -183,10 +190,23 @@ public URI getLocationURI(
 
     /**
      * @since 4.1
+     * @deprecated (4.5.8) use {@link #createLocationURI(String, boolean)}
      */
+    @Deprecated
     protected URI createLocationURI(final String location) throws ProtocolException {
+        return SENTINEL_URI;
+    }
+
+    /**
+     * @since 4.5.8
+     */
+    protected URI createLocationURI(final String location, final boolean normalizeUri) throws ProtocolException {
```

Comment:
@cstamas I think no normalization is required here at this point. The decision whether of not the request URI requires normalization can be deferred until the request execution. This would eliminate the need for `createLocationURI` deprecation, the sentinel URI and lots of other things.

Reply:
Ack, done